lib/src/components/AbstractDialog/__tests__/AbstractDialog.test.tsx

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+
+import { render, fireEvent, waitFor, createEvent } from '@testing-library/react';
+import { act } from 'react-dom/test-utils';
+import { screen } from '@testing-library/dom';
+import userEvent from '@testing-library/user-event';
+
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
+
+import AbstractDialog, { Props } from '../AbstractDialog';
+import NxButton from '../../NxButton/NxButton';
+import NxTooltip from '../../NxTooltip/NxTooltip';
+import useToggle from '../../../util/useToggle';
+
+describe('AbstractDialog', function() {
+  const minimalProps: Props = {
+    children: 'A message inside the dialog element.',
+    onCancel: jest.fn()
+  };
+
+  const quickRender = rtlRender(AbstractDialog, minimalProps);
+  const getDialog = rtlRenderElement(AbstractDialog, minimalProps);
+
+  it('renders children within the dialog element', function() {
+    expect(quickRender({ children: <div data-testid="bar"/> }).getByTestId('bar')).toBeInTheDocument();
+  });
+
+  it('forwards the dialog element ref', function() {
+    const ref = React.createRef<HTMLDialogElement>();
+    render(<AbstractDialog ref={ref} onCancel={() => {}} />);
+    const dialog = screen.getByRole('dialog', { hidden: true });
+    expect(ref.current).toBe(dialog);
+  });
+
+  it('applies the className to the dialog element', function() {
+    expect(getDialog({ className: 'foo' })).toHaveClass('foo');
```

Comment:
We like to test that adding custom classes does not strip off any of the built-in classes.  See `NxToast.test.tsx` for an example

Reply:
> We like to test that adding custom classes does not strip off any of the built-in classes. See `NxToast.test.tsx` for an example

Thanks, we updated the drawer classname test as part of: d89e53eb5c06d82ee732d20acc0f0f7eb7f0cb83

Reply:
> Thanks, we updated the drawer classname test as part of: https://github.com/sonatype/sonatype-react-shared-components/commit/d89e53eb5c06d82ee732d20acc0f0f7eb7f0cb83

That has it comparing its classes to itself.  You need to compare one render that has a custom classname to a different render that does not.

Reply:
> > Thanks, we updated the drawer classname test as part of: [d89e53e](https://github.com/sonatype/sonatype-react-shared-components/commit/d89e53eb5c06d82ee732d20acc0f0f7eb7f0cb83)
> 
> That has it comparing its classes to itself. You need to compare one render that has a custom classname to a different render that does not.

Oh ops: fixed here: 7f2bedb25b5c90ae802fae48cf4478d97dcb06da